### PR TITLE
Bug 1881154: Add AdditionalUnattendContent on Windows machines

### PR DESCRIPTION
**Pull request by LorbusChris**
This commit adds functionality for automatically running
C:\AzureData\CustomData.bin on first boot.
This custom data file is provided via the user data secret,
and contains init scripts.

The Windows Machine Config Operator relies on the user data script being
started automatically on the Windows VM to set up SSH access.

On AWS, it is invoked automatically after provisioning.
This commit is intended to align Azure with that behavior.

I've tested this manually and verified it works.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/azure/services/virtualmachines/virtualmachines.go [hunk 1]
```diff
@@ -126,6 +126,39 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
...
+				{
+					PassName:      "OobeSystem",
+					ComponentName: "Microsoft-Windows-Shell-Setup",
+					SettingName:   "FirstLogonCommands",
+					Content: to.StringPtr(`
+						<FirstLogonCommands>
+							<SynchronousCommand wcm:action="add">
+								<Description>Launch custom data scripts</Description>	
+								<CommandLine>
+									if exist C:\AzureData\CustomData.bin powershell.exe -File C:\AzureData\CustomData.bin
```
Comment (LorbusChris) [inline pkg/cloud/azure/services/virtualmachines/virtualmachines.go:154]: might need `-encodedcommand` here

**Comment by LorbusChris** [comment]
I can confirm this works
/hold cancel

File: pkg/cloud/azure/services/virtualmachines/virtualmachines.go [hunk 2]
```diff
@@ -126,6 +126,40 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 	if compute.OperatingSystemTypes(vmSpec.OSDisk.OSType) == compute.Windows {
 		osProfile.WindowsConfiguration = &compute.WindowsConfiguration{
 			EnableAutomaticUpdates: to.BoolPtr(false),
+			AdditionalUnattendContent: &[]compute.AdditionalUnattendContent{
+				{
+					PassName:      "OobeSystem",
+					ComponentName: "Microsoft-Windows-Shell-Setup",
+					SettingName:   "AutoLogon",
+					Content: to.StringPtr(`<AutoLogon>
+							<Username>` + *osProfile.AdminUsername + `</Username>
```
Comment (enxebre) [inline pkg/cloud/azure/services/virtualmachines/virtualmachines.go:135]: what indentation convention is this following?
Can this leave into a variable?

File: pkg/cloud/azure/services/virtualmachines/virtualmachines.go [hunk 3]
```diff
@@ -126,6 +126,40 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
...
+					`),
+				},
+				{
+					PassName:      "OobeSystem",
+					ComponentName: "Microsoft-Windows-Shell-Setup",
+					SettingName:   "FirstLogonCommands",
+					Content: to.StringPtr(`<FirstLogonCommands>
+							<SynchronousCommand>
+								<Description>Copy user data secret contents to init script</Description>	
+								<CommandLine>cmd /c "copy C:\AzureData\CustomData.bin C:\init.ps1"</CommandLine>
```
Comment (enxebre) [inline pkg/cloud/azure/services/virtualmachines/virtualmachines.go:151]: what indentation convention is this following?
Can this leave into a constant?

**Comment by enxebre** [comment]
The commit message is broken, please follow `UPSTREAM: <carry>: blah` convention

File: pkg/cloud/azure/services/virtualmachines/virtualmachines.go (hunk 2 above)
Comment (LorbusChris) [reply pkg/cloud/azure/services/virtualmachines/virtualmachines.go:135]: I'm not sure this is what you mean:
This is xml with 1 tab indentation. 

Could you elaborate on the second question?

**Comment by LorbusChris** [comment]
> The commit message is broken, please follow `UPSTREAM: <carry>: blah` convention

For now, this is solely a downstream PR. Does it have to be submitted at https://github.com/kubernetes-sigs/cluster-api-provider-azure and be referenced here before merging in this repo?

**Comment by LorbusChris** [comment]
@enxebre looks like the test is still not happy with the commit message. Any idea what's missing?

File: pkg/cloud/azure/services/virtualmachines/virtualmachines.go (hunk 2 above)
Comment (JoelSpeed) [reply pkg/cloud/azure/services/virtualmachines/virtualmachines.go:135]: We could take the XML out of here and make it a constant with the Username and Password as fmt variables and then here we would just do 

```
Content: to.StringPtr(autoLogonFormat, *osProfile.AdminUsername, *osProfile.AdminPassword),
```

Can we do the same with the `FirstLogonCommands` as well? Move to a constant at the top of this file? That one doesn't look like it needs to be a format string though

**Review by JoelSpeed — COMMENTED** [review]
You commit message currently has `UPSTREAM: <carry>:` but I think it needs `UPSTREAM: <carry>: openshift:` as a prefix to satisfy the git history check

**Comment by LorbusChris** [comment]
@enxebre @JoelSpeed thank you for the reviews! I've moved the strings to a constant and fixed up the commits msgs.

**Pull request merged**